src/manifolds/ProbabilitySimplex.jl

```diff
@@ -190,18 +207,34 @@ injectivity_radius(M::ProbabilitySimplex) = 0
 injectivity_radius(M::ProbabilitySimplex, ::AbstractRetractionMethod) = 0
 
 @doc raw"""
-    inner(M::ProbabilitySimplex,p,X,Y)
+    inner(M::ProbabilitySimplex, p, X, Y)
 
 Compute the inner product of two tangent vectors `X`, `Y` from the tangent space $T_pΔ^n$ at
 `p`. The formula reads
 ````math
 g_p(X,Y) = \sum_{i=1}^{n+1}\frac{X_iY_i}{p_i}
 ````
+When `M` includes boundary, we can just skip coordinates where ``p_i`` is equal to 0, see
+Proposition 2.1 in [^AyJostLeSchwachhöfer2017].
+
+[^AyJostLeSchwachhöfer2017]:
+    > N. Ay, J. Jost, H. V. Le, and L. Schwachhöfer, Information Geometry. in Ergebnisse der
+    > Mathematik und ihrer Grenzgebiete. 3. Folge / A Series of Modern Surveys in
+    > Mathematics. Springer International Publishing, 2017.
+    > doi: [10.1007/978-3-319-56478-4](https://doi.org/10.1007/978-3-319-56478-4)
 """
-function inner(::ProbabilitySimplex, p, X, Y)
+function inner(::ProbabilitySimplex{n,closed}, p, X, Y) where {n,closed}
     d = zero(Base.promote_eltype(p, X, Y))
-    @inbounds for i in eachindex(p, X, Y)
-        d += X[i] * Y[i] / p[i]
+    if closed
```

Comment:
Could we maybe do these via dispatch instead of an if? The (small) advantage would be that we could provide two docstrings instead of one.

Reply:
Yes, though then we would have to duplicate initialization of `d` and parts of the docstring so I'm not sure if that's better.

Reply:
Duplication in doc strings is not that bad I think, in code a bit more, sure. So overall I am not 100% we should split it into two function definitions, but we should at least think a bit about it.